pymc_marketing/clv/models/basic.py

```diff
@@ -78,31 +131,120 @@ def _fit_MAP(self, **kwargs):
         trace = MultiTrace([map_strace])
-        return pm.to_inference_data(trace, model=model)
+        idata = pm.to_inference_data(trace, model=model)
+        self.set_idata_attrs(idata)
+        self.idata = idata
+        return self.idata
+
+    @classmethod
+    def load(cls, fname: str):
+        """
+        Creates a ModelBuilder instance from a file,
+        Loads inference data for the model.
+
+        Parameters
+        ----------
+        fname : string
+            This denotes the name with path from where idata should be loaded from.
+
+        Returns
+        -------
+        Returns an instance of ModelBuilder.
+
+        Raises
+        ------
+        ValueError
+            If the inference data that is loaded doesn't match with the model.
+        Examples
+        --------
+        >>> class MyModel(ModelBuilder):
+        >>>     ...
+        >>> name = './mymodel.nc'
+        >>> imported_model = MyModel.load(name)
+        """
+        filepath = Path(str(fname))
+        idata = az.from_netcdf(filepath)
+        dataset = idata.fit_data.to_dataframe()
+
+        model = cls(
+            dataset,
+            model_config=json.loads(idata.attrs["model_config"]),  # type: ignore
+            sampler_config=json.loads(idata.attrs["sampler_config"]),
+        )
+        model.idata = idata
+
+        model.build_model()
+        # All previously used data is in idata.
+
+        return model
+
+    @staticmethod
+    def _check_prior_ndim(prior, ndim=0):
+        if prior.ndim != ndim:
+            raise ValueError(
+                f"Prior variable {prior} must be have {ndim} ndims, but it has {prior.ndim} ndims."
+            )
+
+    def create_distribution_from_prior(self, name: str, **kwargs) -> TensorVariable:
+        try:
+            prior = getattr(pm, name).dist(**kwargs)
+        except AttributeError:
+            raise ValueError(f"Distribution {name} does not exist in PyMC")
+        return prior
+
+    @staticmethod
+    def _process_priors(
+        *priors: TensorVariable, check_ndim: bool = True
+    ) -> Tuple[TensorVariable, ...]:
+        """Check that each prior variable is unique and attach `str_repr` method."""
+        if len(priors) != len(set(priors)):
+            raise ValueError("Prior variables must be unique")
+        # Related to https://github.com/pymc-devs/pymc/issues/6311
+        for prior in priors:
+            if check_ndim:
+                CLVModel._check_prior_ndim(prior)
+            prior.str_repr = types.MethodType(str_for_dist, prior)  # type: ignore
+        return priors
+
+    @property
+    def default_sampler_config(self) -> Dict:
+        return {}
 
     @property
-    def fit_result(self):
-        if self._fit_result is None:
+    def prior_predictive(self) -> az.InferenceData:
+        if self.idata is None or "prior_predictive" not in self.idata:
             raise RuntimeError("The model hasn't been fit yet, call .fit() first")
```

Comment:
Model doesn't need to be fit in order to do prior_predictive

Reply:
Also make sure to test at least in a dummy subclass

Reply:
in this case the implementation wasn't because of mathematical requirements for obtaining prior_predictive samples, but because the mcmc fit called 'sample_model' function, which did all 3 samplings, prior_predictive, posterior and posterior_predictive. So in case that the self.idata was not set, or the prior_predictive was not found in idata was only because none of the samplings were done ( at least from the API perspective). I can change it as anyway now sampling from prior predictive will need to be called separately

Reply:
and it's already tested, all added properties and functions are 100% test covered

Reply:
The codecov bot here on github showed this one as not being tested, that's why I asked. Not sure how reliable it is though. Skimming through the tests, I didn't see a call to `sample_prior_predictive` directly

Reply:
that's because prior to re-defining the sample_model again for CLV classes (as you said , it shouldn't call by default sample_prior_predictive here) it was always done with fit, I've added dedicated sample_prior_predictive method in base.py and it's corresponding test inside test_basic.py (coverage back to 100%)